docs/src/main/asciidoc/spring-cloud-square.adoc

```diff
@@ -0,0 +1,163 @@
+
+You can also disable OkHttpClient load-balancing via properties, by setting the value of `okhttp.loadbalancer.enabled` to `false`.
+
+== Retrofit integration
+
+We provide Spring Boot and Spring Cloud LoadBalancer integration for https://square.github.io/okhttp/[Retrofit], which is a declarative HTTP client.
+
+In order to enable and instantiate Retrofit clients, use the `@EnableRetrofitClients` annotation over a Spring `@Configuration` class:
+
+[source,java]
+----
+@EnableRetrofitClients(clients = TestClient.class, defaultConfiguration = LoggingRetrofitConfig.class)
+@Configuration
+class AppConfiguration{
+
+}
+----
+
+It scans for interfaces that declare they are clients (via `@RetrofitClient`. You can further control the scope of the scan by specifying packages or client names in the annotation parameters. You can also use the `defaultConfiguration` parameter to specify the default config class for all your Retrofit clients.
+
+You can create load-balanced Retrofit client beans by annotating a Retrofit interface with `@RetrofitClient`.
+
+See two examples bellow, one using `OkHttpClient`-specific `Call` interface in method signatures and the other one that can be used with a `WebClient`-backed implementation:
+
+[source,java]
+----
+@RetrofitClient("serviceId")
+interface HelloClient {
+	@GET("/")
+	Call<String> hello();
+}
+----
+
+[source,java]
+----
+@RetrofitClient("serviceId")
+interface HelloClient {
+	@GET("/")
+	Mono<String> hello();
+}
+----
+
+`@RetrofitClient` takes a `value` (alias `name`) argument, that also indicates the `serviceId` which should be used during load-balancing.
+
+If you do not wish for the calls to be load-balanced, you should also use the `url` parameter, that allows you to set the full url, with host and port specified explicitly:
+
+[source,java]
+----
+@RetrofitClient(name = "localapp", url = "http://localhost:8080")
+	protected interface TestClient {
+
+		@GET("/hello")
+		Call<Hello> getHello();
+}
+----
+
+Apart from this, `@RetrofitClient` annotation allows you to specify a `qualifier` and pass a custom `configuration` class that you can use to provid or override OkHttp `Interceptor` beans, as in the following example:
+
+[source,java]
+----
+@RetrofitClient(name = "localapp", configuration = CustomRetrofitConfig.class)
+	protected interface TestClient {
+
+		@GET("/hello")
+		Call<Hello> getHello();
+}
+
+class CustomRetrofitConfig {
+
+		@Bean
+		public Interceptor interceptor1() {
+			return chain -> {
+				Request request = chain.request().newBuilder().addHeader(MYHEADER1, "myheader1value").build();
+				return chain.proceed(request);
+			};
+		}
+}
+----
+
+TIP: The configurations passed via `@RetrofitClient` and `@EnableRetrofitClients` annotations, behave similarly to the ones passed via `@LoadBalancerClient` and `@LoadBalancerClients` annotations. They are loaded into separate child contexts and should not be annotated with `@Configuration`.
```

Comment:
Change to:

NOTE: The configurations passed by the `@RetrofitClient` and `@EnableRetrofitClients` annotations behave similarly to the ones passed by the `@LoadBalancerClient` and `@LoadBalancerClients` annotations. They are loaded into separate child contexts and should not be annotated with `@Configuration`.